### ISSUE TITLE
Means to set bind_dn and bind_passwd for LDAP?

### ISSUE THREAD

**Issue by firthmj** [nilsteampassnet/TeamPass#2079]
Is there a way to set the bind_dn and bind_passwd fields for LDAP authentication?

My understanding (possibly incorrect) is that the default for connecting to Windows Active Directory is that you need to configure these to a valid user.

### Steps to reproduce
1. Browse to Settings->LDAP
2. Select "Windows / Active Directory" as the LDAP server type

### Expected behaviour
If the above settings are made, should boxes appear to allow the bind_dn and bind_passwd to be set?

### Actual behaviour
When trying to connect to an Active Directory LDAP server with LDP debugging enabled, the debug log shows the bind_dn and bind_passwd fields as empty.

The "After authenticate" result is showing "Can't contact LDAP server", and the "ldap status" is blank

It is possible that other settings are incorrect - I haven't yet found a simple "how to configure TeamPass LDAP for default configured Active Directory" guide.

### Server configuration
**Operating system**: Debian 9

**Web server**: Apache 2 (default Debian 9 version)

**Database**: MariaDB (default Debian 9 version)

**PHP version** 7.0 (default Debian 9 install)

**Teampass version**: 2.1.27.10

**Teampass configuration file:**

**Updated from an older Teampass or fresh install:**
Fresh install

PLEASE attach to this message the file `/includes/config/tp.config.php`.

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### Log from the web-browser developer console (CTRL + SHIFT + i)
```
Insert the log here and especially the answer of the query that failed.
```

**Comment by firthmj** [comment]
I was running with LDAPS enabled. If I turn it off, the "After authenticate" message changes to "Invalid credentials". So it may be I have two problems.

**Comment by nilsteampassnet** [comment]
I think you have something wrong with ldap settings.
Indeed several users have this in place and don't require those 2 more fields.

Perhaps should you share your Teampass LDAP settings so that users may help you.

**Comment by finalbeta** [comment]
Teampass uses the credentials of the user that is loggin into teampass to bind to the LDAP, so it does't need a specific user for this. 
To use LDAPS you will need to use certificates. I suggest to try using LDAP first. 
I did notice that the test button doesn't work for me (just gives "no data"), but if you enter the data correct, the users should be able to log in to teampass from the logon screen.

**Comment by firthmj** [comment]
Hi,

The users are fairly deep in the Active Directory domain, at ou=Users,ou=level2,ou=level1,dc=domain,dc=local

At some point, I will probably want to restrict the group that can log in, and that will be somewhere like ou=Groups,ou=level2,ou=level1,dc=domain,dc=local

(actual names obscured)

So what I currently have for the TeamPass LDAP settings are:

> LDAP server type:                      Windows / Active Directory
> LDAP account suffix:                 @ou=Users,ou=level2,ou=level1,dc=domain,dc=local
> LDAP base DN:                          dc=domain,dc=local
> LDAP allowed user group CN:   (blank)
> LDAP domain controllers:         dc1.domain.local,dc2.domain.local
> LDAP port:                                389

And the next three check box options set to "No"

I am pretty confident that the OUs and domain controllers are correct, as I am using the same values to connect to AD from "nslcd" on some Linux systems.

The main thing that is not clear is what should go in the "LDAP base DN" field - should this always be the same as the "LDAP account suffix", but without the '@'? If so, how will it work if I want to apply a group restriction, as the group would not be within the "Users" DN?

Presumably setting the server type to "Windows" will make it look for the "sAMAccountName" field for the user names, rather than the LDAP standard name?

**Comment by finalbeta** [comment]
this is what I have in my test (start off with one dc, start off at the top level, once that works, go further):

LDAP server type: Windows / Active Directory
LDAP account suffix: @mydmz.local
LDAP base DN: dc=mydmz,dc=local
LDAP allowed user group CN: teampass_allow
LDAP domain controllers: dc1.mydmz.local
LDAP port: 389

**Comment by ethantq** [comment]
This works on Centos and SLES, don't know about debian. Should be similar.
For ldaps on Linux you need to copy your root certs to /etc/pki/trust/anchors and then run update-ca-certificates after that you can setup the ldaps connection.

**Comment by pingax** [comment]
Hi,

I updated this day to last version and were stuck with AD auth like you. I just resolved the issue : in the field "ldap suffix for your domain" (not sure if exact string), try to keep it blank and test auth. Just worked for me (Teampass running on Debian with 2012 Active Directory).


> The main thing that is not clear is what should go in the "LDAP base DN" field - should this always be the same as the "LDAP account suffix", but without the '@'? If so, how will it work if I want to apply a group restriction, as the group would not be within the "Users" DN?

**basedn** is always something like that : `dc=<compagny>,dc=<lan/local/whatever>`. 

If you want to restrict TP's access to one group, just create it in your AD and add members into it. When done, put the group's name in teampass' config : 

![image](https://user-images.githubusercontent.com/51684/38732452-79f2c5f4-3f1e-11e8-9160-89e6ce2a4180.png)
